src/main.cpp

```diff
@@ -1,151 +1,119 @@
 
-void setup()
-{
-    Serial.begin(115200);
+void update_signal(int &key_input) {
+    // l (led)
+    if (key_input == 108) {
+        utils::flash_led(LED_BUILTIN);
+    }
 
-    // Flash LED
-    utils::flash_led(LED_BUILTIN);
+    // b - beep
+    if (key_input == 98) {
+        shoot::throttle_code = 1;
+        shoot::telemetry = 1;
+    }
 
-    // pwm config
-    // Note that PWM needs to be setup first, 
-    // because the dma dreq requires tts::pwm_slice_num
-    tts::pwm_setup();
+    // r - rise
+    if (key_input == 114) {
+        if (shoot::throttle_code >= ZERO_THROTTLE and
+            shoot::throttle_code <= MAX_THROTTLE) {
+            // Check for max throttle
+            if (shoot::throttle_code == MAX_THROTTLE) {
+                printf("Max Throttle reached\n");
+            } else {
+                shoot::throttle_code += 50;
+                if (shoot::throttle_code > MAX_THROTTLE)
+                    shoot::throttle_code = MAX_THROTTLE;
+                printf("Throttle: %i\n", shoot::throttle_code - ZERO_THROTTLE);
+            }
+        } else {
+            printf("Motor is not in throttle mode\n");
+        }
+    }
 
-    // dma config
-    tts::dma_setup();
+    // f - fall
+    if (key_input == 102) {
+        if (shoot::throttle_code <= MAX_THROTTLE &&
+            shoot::throttle_code >= ZERO_THROTTLE) {
+            if (shoot::throttle_code == ZERO_THROTTLE) {
+                printf("Throttle is zero\n");
+            } else {
+                shoot::throttle_code -= 50;
+                printf("Throttle: %i\n", shoot::throttle_code - ZERO_THROTTLE);
+            }
+        } else {
+            printf("Motor is not in throttle mode\n");
+        }
+    }
 
-    // Set repeating timer
-    // NOTE: this can be put in main loop to start 
-    // repeating timer on key press (e.g. a for arm)
-    shoot::rt_setup();
+    // q - disarm
+    // Not sure how to disarm yet. Maybe set throttle to 0 and don't send a
+    // cmd for some secs?
+    if (key_input == 32) {
+        shoot::throttle_code = ZERO_THROTTLE;
+        shoot::telemetry = 0;
+        printf("Throttle: %i\n", 0);
+    }
 
-    delay(1500);
+    shoot::send_dshot_frame();
+    printf("Finished processing byte.\n");
+}
 
-    tts::print_gpio_setup();
-    tts::print_dshot_setup();
-    tts::print_pwm_setup();
-    tts::print_dma_setup();
+int main() {
+    int32_t thrust;
+    int key_input;
 
-    Serial.print("Initial throttle: ");
-    Serial.print(shoot::throttle_code);
-    Serial.print("\tInitial telemetry: ");
-    Serial.println(shoot::telemetry);
-}
+    // Load cell calibration
+    int32_t zero_val = 150622;
 
-int incomingByte;
+    stdio_init_all();
+    gpio_init(LED_BUILTIN);
+    gpio_set_dir(LED_BUILTIN, GPIO_OUT);
 
-void loop()
-{
-    if (Serial.available() > 0)
-    {
-        incomingByte = Serial.read();
-        Serial.print("Byte: ");
-        Serial.println(incomingByte, DEC);
+    for (int i = 0; i < 3; i++) {
```

Comment:
Can this be replaced by `lib/utils::flash_led()`?

Reply:
Done

Reply:
k. Not a fan of non-one liners, so I implemented a beefed up version in [`1e8f21a`](https://github.com/Guppy16/pico-tts/pull/2/commits/1e8f21afed739b6a67fa149ceeb935aa2cb57d7b)!